syntax = "proto3";
package littlehorse;

option go_package = ".;model";
option java_multiple_files = true;
option java_package = "io.littlehorse.sdk.common.proto";
option csharp_namespace = "LittleHorse.Common.Proto";

enum LHStatus {
  STARTING = 0;
  RUNNING = 1;
  COMPLETED = 2;
  HALTING = 3;
  HALTED = 4;
  ERROR = 5;
  EXCEPTION = 6;
}

enum MetadataStatus {
  ACTIVE = 0;
  ARCHIVED = 1;
  TERMINATING = 2;
}

enum TaskStatus {
  TASK_SCHEDULED = 0;
  TASK_RUNNING = 1;
  TASK_SUCCESS = 2;
  TASK_FAILED = 3;
  TASK_TIMEOUT = 4;
  TASK_OUTPUT_SERIALIZING_ERROR = 5;
  TASK_INPUT_VAR_SUB_ERROR = 6;
  TASK_EXCEPTION = 8;
}

// Metrics
enum MetricsWindowLength {
  MINUTES_5 = 0;
  HOURS_2 = 1;
  DAYS_1 = 2;
}

enum VariableType {
  JSON_OBJ = 0;
  JSON_ARR = 1;
  DOUBLE = 2;
  BOOL = 3;
  STR = 4;
  INT = 5;
  BYTES = 6;
}

enum LHErrorType {
  CHILD_FAILURE = 0;
  VAR_SUB_ERROR = 1;
  VAR_MUTATION_ERROR = 2;
  USER_TASK_CANCELLED = 3;
  TIMEOUT = 4;
  TASK_FAILURE = 5;
  VAR_ERROR = 6;
  TASK_ERROR = 7;
  INTERNAL_ERROR = 8;
}

enum WaitForThreadsPolicy{
  STOP_ON_FAILURE = 0;
}
